client.go

```diff
@@ -241,22 +241,79 @@ func (acc *Account) Login(username, password string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode == http.StatusUnauthorized {
+		var errResp ErrorResponse
+		if err := NewDecoder(resp.Body).Decode(&errResp); err != nil {
+			return err
+		}
+		return &errResp
+	}
+
 	var loginResponse LoginResponse
 	if err := NewDecoder(resp.Body).Decode(&loginResponse); err != nil {
 		return err
 	}
-	acc.token = loginResponse.Token
+
+	acc.session = &loginResponse
+	return nil
+}
+
+// PasswordChange changes the password of an active session.
+func (acc *Account) PasswordChange(newPassword string) error {
+	if acc.session == nil {
+		return ErrNoSession
+	}
+	lc, err := acc.loginChallenge(acc.session.User.Username)
+	if err != nil {
+		return err
+	}
+
+	priv := acc.authPriv
+	acc.initKeys(lc.Salt, newPassword)
+
+	encrypedContent, err := crypto.Encrypt(acc.mainKey, append(acc.accountKey, acc.idPriv...))
+	if err != nil {
+		return err
+	}
+
+	req := LoginRequest{
+		Username:         acc.session.User.Username,
+		Challenge:        lc.Challenge,
+		Host:             "api.etebase.com",
```

Comment:
Same about hardcoding the url. Should be saved in the instance.

Reply:
agreed!